src/Voron/Data/Fixed/FixedSizeTree.cs

```diff
@@ -736,13 +748,13 @@ private ushort CopyEmbeddedContentToTempPage(long key, TemporaryPage tmp, out bo
                 header->NumberOfEntries = 1;
 
                 byte* dataStart = ptr + sizeof(FixedSizeTreeHeader.Embedded);
-                *(long*)(dataStart) = key;
+                *(TVal*)(dataStart) = key;
                 return (dataStart + sizeof(long));
```

Comment:
Shouldn't this be `dataStart + sizeof(TVal)`? Won't be a correctness issue but it would waste space if `sizeof(TVal) < sizeof(long)`

Reply:
No, the *size* is always 8 bytes. 

We allow both `long` and `double`, but the actual size must be 8 bytes.